src/components/Dependencies/Dependencies.tsx

```diff
@@ -1,117 +1,77 @@
-    const tagText = `${name}@${version}`;
-    return <Tag className={'dep-tag'} clickable={true} label={tagText} onClick={this.handleOnClick} />;
-  }
+  function handleClick(name: string): void {
+    enableLoading && enableLoading();
 
-  private handleOnClick = () => {
-    const { name } = this.state;
-    const { onLoading, history } = this.props;
-
-    onLoading && onLoading();
     history.push(`/-/web/detail/${name}`);
-  };
-}
-
-const WrapperDependencyDetail = withRouter(DepDetail);
-
-class DependencyBlock extends Component<{ title: string; dependencies: [] }> {
-  public render(): ReactElement<HTMLElement> {
-    const { dependencies, title } = this.props;
-    const deps = Object.entries(dependencies) as [];
-
-    return (
-      <DetailContextConsumer>
-        {({ enableLoading }) => {
-          return (
-            <CardWrap>
-              <CardContent>
-                <Heading variant="subtitle1">{`${title} (${deps.length})`}</Heading>
-                <Tags>{this.renderTags(deps, enableLoading)}</Tags>
-              </CardContent>
-            </CardWrap>
-          );
-        }}
-      </DetailContextConsumer>
-    );
   }
 
-  private renderTags = (deps: [], enableLoading?: () => void) =>
-    deps.map(dep => {
-      const [name, version] = dep as [string, string];
-
-      return <WrapperDependencyDetail key={name} name={name} onLoading={enableLoading} version={version} />;
-    });
+  return (
+    <CardWrap>
+      <CardContent>
+        <Heading variant="subtitle1">{`${title} (${deps.length})`}</Heading>
+        <Tags>
+          {deps.map(([name, version]) => (
+            // eslint-disable-next-line
+            <Tag className={'dep-tag'} clickable={true} key={name} label={`${name}@${version}`} onClick={() => handleClick(name)} />
+          ))}
+        </Tags>
+      </CardContent>
+    </CardWrap>
+  );
+};
+
+function hasKeys(object?: { [key: string]: any }): boolean {
+  return !!object && Object.keys(object).length > 0;
 }
 
-class Dependencies extends Component {
-  public render(): ReactElement<HTMLElement> {
-    return (
-      <DetailContextConsumer>
-        {packageMeta => {
-          return this.renderDependencies(packageMeta as VersionPageConsumerProps);
-        }}
-      </DetailContextConsumer>
-    );
-  }
+const Dependencies: React.FC<{}> = () => {
+  const { packageMeta } = useContext(DetailContext);
 
-  private checkDependencyLength<T>(dependency: Record<string, T> = {}): boolean {
-    return Object.keys(dependency).length > 0;
+  if (!packageMeta) {
+    throw new Error('DetailContext is required');
   }
 
-  private renderDependencies({ packageMeta }): ReactElement<HTMLElement> {
-    const { latest } = packageMeta;
-    const { dependencies, devDependencies, peerDependencies, name } = latest;
+  const { latest } = packageMeta;
+  // FIXME: add dependencies to package meta type
+  // @ts-ignore
+  const { dependencies, devDependencies, peerDependencies, name } = latest;
```

Comment:
We should start working on improving the `PackageMetaInterface` to include all the fields. For now `*dependencies` are not in the type definition